### PR TITLE
[Deps] Migrate to Version Catalogs

### PR DESCRIPTION
**Pull request by ParaskP7**
Project Thread: paaHJt-7cv-p2

This PR migrates all dependencies and plugins in this project to [Version Catalogs](https://docs.gradle.org/current/userguide/platforms.html) and under the newly created [gradle/libs.versions.toml](https://github.com/wordpress-mobile/WordPress-Android/pull/21262/files#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87df) file.

## Description

FYI: I recommend reviewing this PR commit-by-commit just to make sure that every dependency and plugin has been migrated appropriately, just to make sure that there isn't any left-overs that need addressing.

PS: The [gradle/libs.versions.toml](https://github.com/wordpress-mobile/WordPress-Android/pull/21262/files#diff-697f70cdd88ba88fe77eebda60c7e143f6ad1286bca75017421e93ad84fb87df) file, on both, its `versions` and `libraries` sections:
- It is alphabetically ordered.
- It is using the [kebal-case](https://en.wikipedia.org/wiki/Letter_case#Kebab_case) naming convention (or else `dash-case`) .
- All words are lowercased.
- Where it makes sense the `org` name is added in its prefix to group dependencies together and make them easily identifiable.
- Where needed the word `main` is added in its suffix to make sure that a dependency cannot be a parent of another dependency and a dependency itself. This helps in various corner cases.

-----

## To Test:

- Smoke test both, the `Jetpack` and `WordPress` apps, and verify everything is working as expected.
- You could also run the `./gradlew buildHealth` task locally and verify that everything is working as expected ([CI](https://buildkite.com/automattic/wordpress-android/builds/19907)).

-----

## Regression Notes

1. Potential unintended areas of impact

    - Not that I can think of.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

    - See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

    - N/A

-----

## PR Submission Checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

-----

## Testing Checklist (strike-out the not-applying and unnecessary ones): `N/A`

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is larger than 300 lines of changes. Please consider splitting it into smaller PRs for easier and faster reviews.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by nbradbury — COMMENTED** [review]
👍

File: libs/image-editor/build.gradle [hunk 1]
```diff
@@ -43,8 +43,6 @@ repositories {
 }
 
 dependencies {
-    implementation fileTree(dir: 'libs', include: ['*.jar'])
```
Comment (nbradbury) [inline libs/image-editor/build.gradle:46]: 👍

File: build.gradle [hunk 2]
```diff
@@ -24,10 +24,6 @@ ext {
 ext {
     // main
     androidxComposeNavigationVersion = '2.7.7'
-    androidxGridlayoutVersion = '1.0.0'
```
Comment (nbradbury) [inline build.gradle:55]: @ParaskP7 I'm just curious why these three libraries weren't flagged as unused by `buildHealth` when I worked on #21238?

**Review by nbradbury — APPROVED** [review]
@ParaskP7 Fantastic work on this! It took me a bit but I went commit by commit and didn't find any issues. I'll approve but perhaps @wzieba wants to take a look before merging? :shipit:

**Review by wzieba — APPROVED** [review]
Looks good. I'm especially glad that the runtime dependencies are not change by this PR, which proves a thoughtful migration. 👏 

About build classpath changes - they're expected, as we now declare (but not apply) more plugins in the root project. I'm okay with this and I think this is even better as some plugins authors expect us to do this, e.g. AGP ([docs](https://developer.android.com/build/releases/gradle-plugin#updating-plugin)).

File: build.gradle (hunk 2 above)
Comment (ParaskP7) [reply build.gradle:55]: 👋 @nbradbury !

These variables you see there are just versions that could be potentially used (were used in the past) on a dependency/library. When there is no dependency associated with those, there is nothing `buildHealth` or :dependabot: can do identify those "unused" version related variables to report them as such.

Does that make sense? 🙏

**Comment by ParaskP7** [comment]
Awesome, thank you both @nbradbury and @wzieba for the review, to version catalogs and beyond! 🎉 ❤️ 🚀

**Comment by ParaskP7** [comment]
FYI: I just noticed yesterday a missing `main` suffix on the `androidx-constraintlayout` version, so I just added it in this c8240a7 commit and when CI finishes I'll proceed with merging this PR.
